contracts/misc/RewardSplitter.sol

```diff
@@ -124,6 +146,10 @@ contract RewardSplitter is IRewardSplitter, Initializable, OwnableUpgradeable, M
 
   /// @inheritdoc IRewardSplitter
   function claimVaultTokens(uint256 rewards, address receiver) external override {
+    if (rewards == type(uint256).max) {
+      syncRewards();
+      rewards = rewardsOf(msg.sender);
+    }
```

Comment:
You're calling syncRewards twice. Here and in _withdrawRewards - that's extra gas cost